pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go

```diff
@@ -32,8 +32,11 @@ func AddInputResource(
 	logger *zap.SugaredLogger,
 ) (*buildv1alpha1.Build, error) {
 
-	var gitResource *v1alpha1.GitResource
+	if task.Spec.Inputs == nil {
```

Comment:
do you think it makes sense for `Inputs` to be a pointer? if it wasn't a pointer we wouldn't have to check for `nil` here. (I feel like we're a bit inconsistent with when we use pointers and when we don't across our types - maybe it's fine tho!)

Reply:
Following the [k8s API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#optional-vs-required), its a pointer because its optional and its not an array

Reply:
oh yeeeeah that's right, thanks! ill try to remember that for other types also